packages/aws-sdk-client-mock/src/sinon.ts

```diff
@@ -7,3 +7,7 @@ export interface MaybeSinonProxy {
 export const isSinonStub = (obj: unknown): obj is SinonStub =>
     ((obj as MaybeSinonProxy).isSinonProxy || false)
     && (obj as SinonStub).restore !== undefined;
+
+export const isSinonSpy = (obj: unknown): obj is SinonSpy =>
+    ((obj as MaybeSinonProxy).isSinonProxy || false)
+    && (obj as SinonSpy).restore !== undefined;
```

Comment:
Those methods do not distinct between Sinon Stub and Spy. But we don't really care about the distinction, only whether the object is any Sinon Proxy - Stub or Spy. We can get rid of the `restore` existence check and just make it a single `isSinonProxy()` function.

Reply:
I'll remove the `restore` check